server/data/mappers/createGoalMapper.test.ts

```diff
@@ -31,4 +36,34 @@ describe('createGoalMapper', () => {
     // Then
     expect(actual).toEqual(expectedCreateGoalRequest)
   })
+
+  it('should map to CreateGoalRequest given a valid DTO whose targetCompletionDate is in BST', () => {
+    // Given
+    const createGoalDto: CreateGoalDto = aValidCreateGoalDtoWithMultipleSteps()
+    createGoalDto.targetCompletionDate = parse('2024-08-10', 'yyyy-MM-dd', today) // August is BST
+
+    const expectedAddStepRequest1: CreateStepRequest = {
+      title: createGoalDto.steps[0].title,
+      sequenceNumber: 1,
+    }
+    const expectedAddStepRequest2: CreateStepRequest = {
+      title: createGoalDto.steps[1].title,
+      sequenceNumber: 2,
+    }
+    const expectedCreateGoalRequest: CreateGoalRequest = {
+      prisonNumber: createGoalDto.prisonNumber,
+      title: createGoalDto.title,
+      category: 'WORK',
+      steps: [expectedAddStepRequest1, expectedAddStepRequest2],
+      targetCompletionDate: '2024-08-10',
```

Comment:
Without the change in the implementation (file below), this test fails because the date is formatted as `2024-08-09`